app/src/components/integrations/elements/pluginIcon/pluginIcon.jsx

```diff
@@ -24,13 +24,20 @@ import { globalIntegrationsSelector } from 'controllers/plugins/selectors';
 import { filterIntegrationsByName, isPluginSupportsCommonCommand } from 'controllers/plugins/utils';
 import { PLUGIN_DEFAULT_IMAGE, PLUGIN_IMAGES_MAP } from 'components/integrations/constants';
 import { Image } from 'components/main/image';
+import { PUBLIC_PLUGIN_ACCESS_TYPE } from 'controllers/plugins/constants';
 
 export const PluginIcon = ({ pluginData, className, ...rest }) => {
   const { details, name } = pluginData;
   const isDynamicIconAvailable = details && details.binaryData && details.binaryData.icon;
   const projectId = useSelector(activeProjectSelector);
   const globalIntegrations = useSelector(globalIntegrationsSelector);
 
+  const isPublic = details && details.accessType === PUBLIC_PLUGIN_ACCESS_TYPE;
+  const isCommand =
+    (isPluginSupportsCommonCommand(pluginData, COMMAND_GET_FILE) ||
+      filterIntegrationsByName(globalIntegrations, name)[0]) &&
```

Comment:
Seems like you can calculate it inside of `calculateIconSrc` function to avoid calling these functions twice.
I also suggest to update `calculateIconSrc` function to be able to extract from it not only url, but `requestParams` also.
It will help to avoid [this](https://github.com/reportportal/service-ui/pull/3085/files#diff-aeef3323931ea4f09685d937f8bca6535b7a8066eac3d3ebb4359afd06c8aeafR69) condition.